tests/conftest.py

```diff
@@ -214,20 +214,41 @@ def pytest_addoption(parser):
                      help="Delay between qutebrowser commands.")
     parser.addoption('--qute-profile-subprocs', action='store_true',
                      default=False, help="Run cProfile for subprocesses.")
-    parser.addoption('--qute-bdd-webengine', action='store_true',
-                     help='Use QtWebEngine for BDD tests')
+    parser.addoption('--qute-bdd-backend', action='store',
+                     choices=['webkit', 'webengine'], help='Set backend for BDD tests')
 
 
 def pytest_configure(config):
-    webengine_arg = config.getoption('--qute-bdd-webengine')
-    webengine_env = os.environ.get('QUTE_BDD_WEBENGINE', 'false')
-    config.webengine = webengine_arg or webengine_env == 'true'
-    # Fail early if QtWebEngine is not available
-    if config.webengine:
+    try:
+        # Try to use QtWebkit as the default backend
+        import PyQt5.QtWebKitWidgets
+        config.backend = 'webkit'
```

Comment:
It seems a bit confusing to me when reading the code that this sets `config.backend`, which then gets overridden again below.

Maybe call it `auto_backend` or something?

Reply:
The idea behind using `config.backend` to store the backend's name was to make it available to `pytest_report_header()` and then decide based on `config.backend`'s value what to report.

Considering https://github.com/qutebrowser/qutebrowser/pull/6810#discussion_r747610129 it probably makes more sense to implement some checks based on `config.webengine` in `pytest_report_header()` to report the backend:
```python
def pytest_report_hook(config):
    if config.webengine:
        # webengine
    else:
        # webkit
```